### PR TITLE
fix(shiny-preset): Avoid specificity issues in card cap bg

### PR DESCRIPTION
**Pull request by gadenbuie**
Fixes the issue reported in #925 by setting the card cap bg using CSS variables in the dashboard mode.

After this change either of the first two properties sets the card header background for `card_header(class = "my-card-header")`:

```css
.my-card-header {
  --bs-card-cap-bg: #fc6a03;
  background-color: #fc6a03;
  color: white;
}
```

## Explanation

To clarify what's happening in this PR, we need to provide a dashboard-specific background color for `--bs-card-cap-bg`. Essentially, instead of `--bs-card-cap-bg: transparent` we want `--bs-card-cap-bg: var(--bs-body-bg)`. The reason is that we don't want a transparent card cap bg to pick up the (potentially) gray background in the dashboard area.

So we do two things:

```scss
:root {
  --bslib-dashboard-card-cap-bg: var(--bs-body-bg);
}

.bslib-card {
  --bs-card-cap-bg: var(--bslib-dashboard-card-cap-bg);
}
```

Setting `--bslib-dashboard-card-cap-bg` gives users a way to change our default. We could also have written this:

```scss
.bslib-card {
  --bs-card-cap-bg: var(--bslib-dashboard-card-cap-bg, var(--bs-body-bg));
}
```

but I was following the pattern we used for other dashboard-page-specific variables.

Secondly, we set `--bs-card-cap-bg` in `.bslib-card`, which gives users three ways to change the background color:

1. Globally via `--bslib-dashboard-card-cap-bg`
2. For specific headers using a class applied to `card_header()` (see example above in PR)
3. Set `--bs-card-cap-bg` in `.bslib-card`.

## Example

<details>
<summary>Example app</summary>

```r
library(shiny)
library(bslib)

ui <- page_fixed(
  theme = bs_theme() |>
    bs_add_rules("
      .my-card-header {
         --bs-card-cap-bg: #fc6a03;
         background-color: #fc6a03;
         color: white;
       }
    "),
  card(
    card_header("Header", class = "my-card-header"),
    "Body"
  ),
  card(
    card_header("Shiny Preset Card"),
    lorem::ipsum(1)
  )
)

server <- function(input, output, session) {

}

shinyApp(ui, server)
```

</details>

### CODE REVIEW

File: inst/builtin/bs5/shiny/_rules.scss [hunk 1]
```diff
@@ -115,6 +115,8 @@ $bslib-checkbox-radio-margin-right: 0.35em !default;
     --bslib-dashboard-sidebar-main-bg: var(--#{$prefix}body-bg);
     --bslib-dashboard-main-bg: rgb(247, 247, 247);
     --#{$prefix}card-border-color: var(--bslib-dashboard-border-color-translucent, var(--#{$prefix}border-color-translucent));
+    --bslib-dashboard-card-cap-bg: var(--#{$prefix}body-bg);
```
Comment (cpsievert) [inline inst/builtin/bs5/shiny/_rules.scss:118]: Do you recall the reasoning for changing from `transparent` to `var(--#{$prefix}body-bg)`?

**Comment by cpsievert** [comment]
I think I'm OK with this, but we should really have a serious look at adopting CSS layers

File: inst/builtin/bs5/shiny/_rules.scss (hunk 1 above)
Comment (gadenbuie) [reply inst/builtin/bs5/shiny/_rules.scss:118]: The card might be on a colored background -- like our `.bslib-page-dashboard` area.

**Review by cpsievert — APPROVED** [review]

**Comment by gadenbuie** [comment]
@cpsievert btw, I found this PR hard to parse coming back to it after a long delay so I dug into it a little more and updated the description above. I think the change is good, so I'll merge after checks pass, but let me know if you have any comments.

**Comment by cpsievert** [comment]
Thanks for adding that additional context, I'll go ahead and merge just to make sure we get "near code complete" test results tomorrow
